miniapps/meshing/mesh-optimizer.cpp

```diff
@@ -519,6 +534,10 @@ int main (int argc, char *argv[])
       for (int i = 0; i < mesh->GetNBE(); i++)
       {
          const int attr = mesh->GetBdrElement(i)->GetAttribute();
+         MFEM_VERIFY(!(dim == 2 && attr == 3),
+                     "Boundary attribute 3 must be used only for 3D meshes. "
+                     "Adjust the attributes (1/2/3 for fixed x/y/z components, "
+                     "4 for free nodes), or use the option -fix-bnd.");
```

Comment:
Actually, boundary attribute 4 fixes all components; boundary attribute 5 or larger will leave the nodes free.

Reply:
Fixed